### PR TITLE
feat: add e2e test case for yurtstaticset controller

### LINKED ISSUE #1477: [feature request]add e2e test case for yurtstaticset controller
**What would you like to be added**:
add e2e test case for yurtstaticset controller

**Why is this needed**:
improve e2e test coverage

**others**
/kind feature

### PR DESCRIPTION
**Pull request by y-ykcir**
#### Which issue(s) this PR fixes:

Fixes #1477 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rambohe-ch** [comment]
@xavier-hou PTAL

File: test/e2e/yurt/yurtstaticset.go [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+	yurtStaticSetName := "busybox"
+	podName := "busybox"
+	testContainerName := "bb"
+	testImg1 := "busybox"
+	testImg2 := "busybox:1.36.0"
+
+	createNamespace := func() {
+		ns := corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
```
Comment (vie-serendipity) [inline test/e2e/yurt/yurtstaticset.go:59]: Is it more reasonable to use Namespace filed? (including client.ObjectKey below)
```suggestion
				Namespace: namespaceName,
```

File: test/e2e/yurt/yurtstaticset.go [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+	AfterEach(func() {
+		By("Cleanup resources after test")
+		deleteStaticPod("openyurt-e2e-test-worker")
+		deleteStaticPod("openyurt-e2e-test-worker2")
+
+		By(fmt.Sprintf("Delete the entire namespaceName %s", namespaceName))
+		Expect(k8sClient.Delete(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}, client.PropagationPolicy(metav1.DeletePropagationBackground))).Should(BeNil())
+	})
+
+	Describe("Test YurtStaticSet auto upgrade model", func() {
```
Comment (xavier-hou) [inline test/e2e/yurt/yurtstaticset.go:256]: auto -> AdvancedRollingUpdate

File: test/e2e/yurt/yurtstaticset.go [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+		It("Test one worker disconnect", func() {
+			By("Run staticpod auto upgrade model test")
```
Comment (xavier-hou) [inline test/e2e/yurt/yurtstaticset.go:258]: same

**Comment by xavier-hou** [comment]
@y-ykcir  Could you add e2e test case for this new feature? 
https://github.com/openyurtio/openyurt/pull/1516

File: test/e2e/yurt/yurtstaticset.go (hunk 1 above)
Comment (y-ykcir) [reply test/e2e/yurt/yurtstaticset.go:59]: Sorry for not understanding your suggestion. The namespace is a cluster scope resource, and its namespace field will be ignored

**Comment by y-ykcir** [comment]
> @y-ykcir Could you add e2e test case for this new feature? #1516

Done. cc@xavier-hou

**Comment by xavier-hou** [comment]
@rambohe-ch  @y-ykcir  lgtm

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**